drv/lpc55-swd/src/main.rs

```diff
@@ -715,44 +711,40 @@ impl NotificationHandler for ServerImpl {
                     }
                 }
                 Err(e) => {
-                    // This should only fail if JTAG_DETECT or SP_RESET are currently asserted.
+                    // This should only fail if JTAG_DETECT or SP_RESET are
+                    // currently asserted.
                     ringbuf_entry!(Trace::TimerHandlerError(e));
                 }
             }
         }
 
         if (bits & notifications::SP_RESET_IRQ_MASK) != 0 {
-            ringbuf_entry!(Trace::SpResetFired);
-            if !invalidate && !self.do_handle_sp_reset() {
-                ringbuf_entry!(Trace::InvalidateSpMeasurement);
-                // Clear the attestation log
-                invalidate = true;
-            }
-            self.next_use_must_setup_swd();
-
-            //  We are not going to try to measure/trust the SP
-            //  when there is a glitch on the JTAG_DETECT signal.
+            //  We are not going to try to measure/trust the SP when there is a
+            //  glitch on the JTAG_DETECT signal.
             //
             //  e.g. JTAG_DETECT fired but before the handler was called, it
             //  de-asserted so that the SP_RESET that also fired could be
             //  handled successfully.
-            //}
+            ringbuf_entry!(Trace::SpResetFired);
+            if matches!(self.state, SwdState::Connected { .. })
+                && !self.do_handle_sp_reset()
+            {
+                // If handling the SP reset failed, clear the attestation log
+                ringbuf_entry!(Trace::InvalidateSpMeasurement);
+                let _ = self.invalidate_sp_measurement();
```

Comment:
previously there was a comment here saying that `invalidate_sp_measurement()` logs to the ringbuf: https://github.com/oxidecomputer/hubris/pull/2228/files#diff-703eba1954e126d581c58dcb778473d2547077100facfcf04514b30ef3fc2c35L764-L768

IIUC that function hasn't changed here, is that still the case?